ecosystem/indexer-grpc/indexer-grpc-utils/src/storage_format.rs

```diff
@@ -0,0 +1,789 @@
+// Copyright © Aptos Foundation
+
+use anyhow::Context;
+use aptos_protos::{indexer::v1::TransactionsInStorage, transaction::v1::Transaction};
+use bzip2::read::{BzDecoder, BzEncoder};
+use flate2::read::{GzDecoder, GzEncoder};
+use prost::Message;
+use ripemd::{Digest, Ripemd128};
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, io::Read};
+
+const FILE_ENTRY_TRANSACTION_COUNT: u64 = 1000;
+
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq)]
+pub enum StorageFormat {
+    Bz2CompressedProto,
+    GzipCompressionProto,
+    UncompressedProto,
+    // Only used for legacy file format.
+    // Use by cache only.
+    Base64UncompressedProto,
+    // Only used for legacy file format.
+    // Use by file store only.
+    JsonBase64UncompressedProto,
+}
+
+pub enum CacheEntry {
+    Bz2CompressedProto(Vec<u8>),
+    GzipCompressionProto(Vec<u8>),
+    // Only used for legacy cache entry.
+    Base64UncompressedProto(Vec<u8>),
+}
+
+// into_inner to get the inner Vec<u8>
+impl CacheEntry {
+    pub fn into_inner(self) -> Vec<u8> {
+        match self {
+            CacheEntry::Bz2CompressedProto(bytes) => bytes,
+            CacheEntry::GzipCompressionProto(bytes) => bytes,
+            CacheEntry::Base64UncompressedProto(bytes) => bytes,
+        }
+    }
+
+    pub fn size(&self) -> usize {
+        match self {
+            CacheEntry::Bz2CompressedProto(bytes) => bytes.len(),
+            CacheEntry::GzipCompressionProto(bytes) => bytes.len(),
+            CacheEntry::Base64UncompressedProto(bytes) => bytes.len(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct TransactionsLegacyFile {
+    /// The version of the first transaction in the blob.
+    pub starting_version: u64,
+    /// The transactions in the blob.
+    #[serde(rename = "transactions")]
+    pub transactions_in_base64: Vec<String>,
+}
+
+/// FileStoreMetadata is the metadata for the file store.
+/// It's a JSON file with name: metadata.json.
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Eq)]
+pub struct FileStoreMetadata {
+    pub chain_id: u64,
+    // The size of each file folder, BLOB_STORAGE_SIZE, i.e., 1_000.
+    pub file_folder_size: usize,
+    // The current version of the file store.
+    pub version: u64,
+    // Storage format.
+    #[serde(default = "default_storage_format")]
+    pub storage_format: StorageFormat,
+}
+
+fn default_storage_format() -> StorageFormat {
+    StorageFormat::JsonBase64UncompressedProto
+}
```

Comment:
the default cache and file store storage formats are defined in so many places. can we make it cleaner and define the functions once, then use it everywhere?